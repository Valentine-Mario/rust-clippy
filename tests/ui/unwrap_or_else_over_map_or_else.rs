#![warn(clippy::unwrap_or_else_over_map_or_else)]

fn main() {
    let c = func_result(3).unwrap_or_else(|e| {e});
    func_result(2).map_or_else(|e| println!("{:?}", e), |n| println!("{}", n))
}

fn func_result(in_num: u8) -> Result<&'static str, &'static str> {
    if in_num % 2 != 0 {
        return Err("Can't do this because input is odd...");
    }
    Ok("An even number :)")
}
