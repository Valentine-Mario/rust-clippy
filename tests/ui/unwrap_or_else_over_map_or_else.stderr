error: Result::unwrap_or_else is shorter and more succinet
  --> $DIR/unwrap_or_else_over_map_or_else.rs:9:20
   |
LL |     func_result(2).map_or_else(|e| println!("{:?}", e), |n| println!("{}", n))
   |                    ^^^^^^^^^^^
   |
   = note: `-D clippy::unwrap-or-else-over-map-or-else` implied by `-D warnings`
   = help: consider unwrap_or_else(|e| handle_the_error(e)) to unpack result

error: aborting due to previous error

